packages/documentation/copy/en/handbook-v2/Type Manipulation/Template Literal Types.md

```diff
@@ -216,3 +216,39 @@ function applyStringMapping(symbol: Symbol, str: string) {
     return str;
 }</pre></code>
 </details>
+
+### Non-string Types in Template Literals
+
+You can use the types `boolean`, `null`, `undefined`, `number`, and `bigint` in template literal types.
+`${boolean}` is equivelant to `"true" | "false"`, `${null}` and `${undefined}` match the strings `null` and `undefined` respectivly. 
+`${number}` and `${bigint}` match values that the JavaScript will parse to finite numbers. 
+
+```ts twoslash
+let number: `${number}`
+// These are all assignable to `${number}`
+number = '1' // decimal
+number = '0b10' // binary
+number = '0x10' // hexidecimal
+number = '1.0' // floating point
+number = '1e10' // scientific notation
+number = '-1' // negative numbers
+number = ' 1 ' // numbers in whitespace
+number = ' ' // just whitespace
+
+// These are errors because the runtime does not parse them to finite numbers.
+number = "NaN"
+number = "Infinity"
+number = "1_000" // 1_000 is a literal number but parses as NaN.
+
+let bigint: `${bigint}`
+// These are all assignable to `${bigint}`
+bigint = '1' // decimal
+bigint = '0b10' // binary
+bigint = '0x10' // hexidecimal
+bigint = '-1' // negative numbers
+
+// These are errors because the runtime does not parse them to finite numbers.
+bigint = "NaN"
+bigint = "Infinity"
+bigint = "1000n" // 1000n is a literal BigInt but is not parsed as a BigInt.
```

Comment:
I left this out of the examples but `${bigint}` _doesnt_ allow whitespace while `${number}` does, despite BigInt(' ') parsing to 0n. That breaks the original rule as I understood it. Is that a bug in typescript?